src/Uno.UI/UI/Xaml/BorderHelper.skia.cs

```diff
@@ -47,31 +47,42 @@ public static void UpdateAllBorderProperties(this IBorderInfoProvider @this)
 		@this.UpdateBorderThickness();
 	}
 
-	public static void SetUpBrushTransitionIfAllowed(BorderVisual visual, Brush fromBrush, Brush toBrush, BrushTransition transition)
+	public static void SetUpBrushTransitionIfAllowed(BorderVisual visual, Brush fromBrush, Brush toBrush, BrushTransition transition, bool isAnimation)
 	{
 		if (transition is null)
 		{
 			return;
 		}
 
-		var oldBrush = fromBrush as SolidColorBrush;
-		var newBrush = toBrush as SolidColorBrush;
+		if (isAnimation)
+		{
+			// When an animation sets the background, it skips the transition logic and is applied immediately.
+			// However, it "deactivates" the currently active transition if one exists. Once the animation is done
+			// the deactivated transition is reactivated and continues as if it was running (i.e. it's not paused,
+			// but takes into account the duration during which it was deactivated).
+			visual.Compositor.DeactivateBackgroundTransition(visual);
```

Comment:
Could we add a `return;` after this line and remove the else block?


Also, this method was a port from WinUI, so unclear why we need an Uno-specific workaround